docs/developer/decision-records/2023-02-28-processing-callbacks/README.md

```diff
@@ -0,0 +1,147 @@
+# Processing Callbacks
+
+## Decision
+
+Implement the ability for clients to specify callback endpoints to be invoked at specific points during request processing:
+
+- Callback endpoints can be specified for _Contract Negotiation_ and _Transfer Process_ state transitions when a request is made via the _Management API_
+- Callback endpoints are scoped to the particular state machine instance associated with the request.
+- Clients can request the callbacks to be invoked *transactionally* or on a *best-effort* basis.
+- Callback endpoints can be remote or local (in-process). The initial implementation will support HTTPS endpoints.
+
+## Rationale
+
+Currently, the EDC provides support for listeners (e.g. `TransferProcessListener`) and event subscribers (`EventSubscriber`). However, there is no way to provide callback endpoint
+information that can be used by listener and subscriber implementations when a request is made via the Management API. Processing callbacks will address this gap and provide
+support for invoking HTTPS endpoints.
+
+> Note, this functionality can also be used to align EDR handling with the migration to the IDS Dataspace Protocol specifications.
+
+## Configuration Scheme
+
+Support for two types of callbacks will be implemented: static endpoints and dynamic endpoints. Static endpoints are configured in the EDC and referenced by identifier; dynamic
+endpoints are specified at the time a request is made.
+
+### Static Endpoints
+
+The static endpoint registration will be done using a configuration extension. This allows for the development of more elaborate endpoint registration mechanisms in the future.
+
+A static endpoint will be registered using EDC configuration:
+
+```
+edc.callback.endpoint1.uri=
+edc.callback.endpoint1.context=
+edc.callback.endpoint1.states=
+edc.callback.endpoint1.transactional=
+edc.callback.endpoint1.auth-key=
+edc.callback.endpoint1.auth-code-id=
+```
+
+The previous configuration will create a static endpoint registration for the id `endpoint1` with the following properties:
+
+- The `uri` is scheme-specific address used to dispatch a callback notification.
+- The `context` is: `contract-negotiation` or `transfer-process`
+- The optional `states` property is a collection of processing states corresponding to a state machine state. If absent, the default value is all states. The '*' character may be
```

Comment:
Why don't associate this with events instead of states? this way context will be not necessary.

Reply:
+1